### PR TITLE
Fix for HTTPS proxy

### PR DESCRIPTION
**Pull request by jademackay**
Instantiate a new `target_url` specifying the port (443) for https proxy.

### CODE REVIEW

File: src/ConnectionRequest.jl [hunk 1]
```diff
@@ -82,6 +82,9 @@ function request(::Type{ConnectionPoolLayer{Next}}, url::URI, req, body;
 
     try
         if proxy !== nothing && target_url.scheme == "https"
+            target_url = URI(scheme=target_url.scheme,
```
Comment (quinnj) [inline src/ConnectionRequest.jl]: I think we could just do `target_url = merge(target_url, port=443)`, right? That way we keep the rest of the uri components?

**Comment by quinnj** [comment]
Sorry for the slowness in responding here; I'm trying to follow up on things now. I left a comment above.

File: src/ConnectionRequest.jl (hunk 1 above)
Comment (jademackay) [reply src/ConnectionRequest.jl]: Yes, that is better.
I have made the change and successfully executed `HTTP.request("GET", "https://httpbin.org/ip")`.

**Comment by jademackay** [comment]
On the contrary, I really appreciate all the work you're doing. 👍

**Pull request merged**